packages/kbn-ebt-tools/src/metric_events/schema.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { RootSchema } from '@kbn/analytics-client';
+
+/**
+ * Structure of the `metric` event
+ */
+export interface MetricEvent {
+  /**
+   * The name of the action that is tracked in the metrics.
+   */
+  eventName: string;
+  /**
+   * Searchable but not aggregateable metadata relevant to the tracked action.
+   */
+  meta?: Record<string, unknown>;
+
+  /**
+   * @group Standardized fields
+   * The time (in milliseconds) it took to run the entire action.
+   */
+  duration: number;
+
+  /**
+   * @group Free fields for custom metrics (searchable and aggregateable)
+   * Description label for the metric 1
+   */
+  key1?: string;
+  /**
+   * @group Free fields for custom metrics (searchable and aggregateable)
+   * Value for the metric 1
+   */
+  value1?: number;
+  /**
+   * @group Free fields for custom metrics (searchable and aggregateable)
+   * Description label for the metric 2
+   */
+  key2?: string;
+  /**
+   * @group Free fields for custom metrics (searchable and aggregateable)
+   * Value for the metric 2
+   */
+  value2?: number;
+  /**
+   * @group Free fields for custom metrics (searchable and aggregateable)
+   * Description label for the metric 3
+   */
+  key3?: string;
+  /**
+   * @group Free fields for custom metrics (searchable and aggregateable)
+   * Value for the metric 3
+   */
+  value3?: number;
+  /**
+   * @group Free fields for custom metrics (searchable and aggregateable)
+   * Description label for the metric 4
+   */
+  key4?: string;
+  /**
+   * @group Free fields for custom metrics (searchable and aggregateable)
+   * Value for the metric 4
+   */
+  value4?: number;
+  /**
+   * @group Free fields for custom metrics (searchable and aggregateable)
+   * Description label for the metric 5
+   */
+  key5?: string;
+  /**
+   * @group Free fields for custom metrics (searchable and aggregateable)
+   * Value for the metric 5
+   */
+  value5?: number;
+}
+
+export const METRIC_EVENT_SCHEMA: RootSchema<MetricEvent> = {
+  eventName: {
+    type: 'keyword',
+    _meta: { description: 'Type of the event' },
```

Comment:
Nit: Should this rather say "The name of the action that is tracked in the metrics." (to match the API doc comment in `MetricEvent` above)?

Reply:
I don't think we really want to change this description

Reply:
I was pointing out that the API doc of the corresponding `MetricEvent` interface and the schema are inconsistent. So we either need to change the schema or the API docs in order for them to be consistent. Does this make sense now? :)